### PR TITLE
Add ONNX Runtime training benchmark

### PR DESCRIPTION
**Pull request by JingyaHuang**
# What does this PR do?

- [x] Add benchmark results of ONNX Runtime Training to the doc.
- [x] Insert link to the ORT training blog post.

### CODE REVIEW

File: docs/source/onnxruntime/usage_guides/trainer.mdx [hunk 1]
```diff
@@ -10,14 +10,43 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
 
-# Trainer
+# How to accelerate training with ONNX Runtime
 
-The [`ORTTrainer`] and [`ORTSeq2SeqTrainer`] classes provide APIs for training PyTorch models with __[ONNX Runtime (ORT)](https://onnxruntime.ai/)__.
-Taking ONNX Runtime as backend, [`ORTTrainer`] and [`ORTSeq2SeqTrainer`] optimize the computation graph and the memory usage. They also support
-mixed precision training implemented by ORT, as well as distributed training on multiple GPUs. With them, you will be able to achieve
-__lower latency, higher throughput, and larger maximum batch size__ while training large transformers models.
+Optimum integrates ONNX Runtime Training through an `ORTTrainer` API that extends `Trainer` in [Transformers](https://huggingface.co/docs/transformers/index).
+With the solution, training time can be improved by greater than 35% for many popular Hugging Face models compared to PyTorch under eager mode.
```
Comment (fxmarty) [inline docs/source/onnxruntime/usage_guides/trainer.mdx:16]:
```suggestion
With this extension, training time can be reduced by more than 35% for many popular Hugging Face models compared to PyTorch under eager mode.
```
just taste

File: docs/source/onnxruntime/usage_guides/trainer.mdx [hunk 2]
```diff
@@ -10,14 +10,43 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
 
-## Prerequisite
+[`ORTTrainer`] and [`ORTSeq2SeqTrainer`] APIs make it easy to compose __[ONNX Runtime (ORT)](https://onnxruntime.ai/)__ with other features in [`Trainer`](https://huggingface.co/docs/transformers/main_classes/trainer).
+It contains feature-complete training loop and evaluation loop, and supports hyperparameter search, mixed-precision training and distributed training with multiple [NVIDIA](https://techcommunity.microsoft.com/t5/ai-machine-learning-blog/accelerate-pytorch-transformer-model-training-with-onnx-runtime/ba-p/2540471)
+and [AMD](https://cloudblogs.microsoft.com/opensource/2021/07/13/onnx-runtime-release-1-8-1-previews-support-for-accelerated-training-on-amd-gpus-with-the-amd-rocm-open-software-platform/) GPUs.
+With ONNX Runtime backend, [`ORTTrainer`] and [`ORTSeq2SeqTrainer`] take advantage of:
```
Comment (fxmarty) [inline docs/source/onnxruntime/usage_guides/trainer.mdx:21]:
```suggestion
Through the ONNX Runtime Training backend, [`ORTTrainer`] and [`ORTSeq2SeqTrainer`] take advantage of:
```

File: docs/source/onnxruntime/usage_guides/trainer.mdx [hunk 3]
```diff
@@ -10,14 +10,43 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+
+* Computation graph optimizations: node fusions, node eliminations...
```
Comment (fxmarty) [inline docs/source/onnxruntime/usage_guides/trainer.mdx:23]:
```suggestion
* Computation graph optimizations: node fusions, node eliminations.
```

File: docs/source/onnxruntime/usage_guides/trainer.mdx [hunk 4]
```diff
@@ -10,14 +10,43 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+* Efficient memory planning
+* Kernel optimization
+* ORT fused Adam optimizer: batches the elementwise updates applied to all the model's parameters into one or a few kernel launches
+* More efficient FP16 optimizer: eliminates a great deal of device to hos memory copies
```
Comment (fxmarty) [inline docs/source/onnxruntime/usage_guides/trainer.mdx:27]:
```suggestion
* More efficient FP16 optimizer: eliminates a great deal of device to host memory copies
```

File: docs/source/onnxruntime/usage_guides/trainer.mdx [hunk 5]
```diff
@@ -10,14 +10,43 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+* Mixed precision training
+* ...
+
+Test it out to achieve __lower latency, higher throughput, and larger maximum batch size__ while training models in 🤗Transformers!
```
Comment (fxmarty) [inline docs/source/onnxruntime/usage_guides/trainer.mdx:31]:
```suggestion
Test it out to achieve __lower latency, higher throughput, and larger maximum batch size__ while training models in 🤗 Transformers!
```

File: docs/source/onnxruntime/usage_guides/trainer.mdx [hunk 6]
```diff
@@ -10,14 +10,43 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+
+## Performance
+
+The chart shows impressive acceleration __from 39% to 130%__ for Hugging Face models with Optimum when __using ONNX Runtime and DeepSpeed ZeRO Stage 1__ for training.
```
Comment (fxmarty) [inline docs/source/onnxruntime/usage_guides/trainer.mdx:35]:
```suggestion
The chart below shows impressive acceleration __from 39% to 130%__ for Hugging Face models with Optimum when __using ONNX Runtime and DeepSpeed ZeRO Stage 1__ for training.
```

File: docs/source/onnxruntime/usage_guides/trainer.mdx [hunk 7]
```diff
@@ -10,14 +10,43 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+The performance measurements were done on selected Hugging Face models with PyTorch as the baseline run, only ONNX Runtime for training as the second run, and ONNX
+Runtime + DeepSpeed ZeRO Stage 1 as the final run, showing maximum gains. The Optimizer used for the baseline PyTorch runs is the AdamW optimizer and the ORT Training
+runs use the Fused Adam Optimizer. The runs were performed on a single Nvidia A100 node with 8 GPUs.
```
Comment (fxmarty) [inline docs/source/onnxruntime/usage_guides/trainer.mdx:38]:
```suggestion
Runtime + DeepSpeed ZeRO Stage 1 as the final run, showing maximum gains. PyTorch baseline runs use AdamW optimizer, while the ORT Training runs use the Fused Adam optimizer.
```

is there a reference for fused adam?

File: docs/source/onnxruntime/usage_guides/trainer.mdx (hunk 2 above)
Comment (JingyaHuang) [reply docs/source/onnxruntime/usage_guides/trainer.mdx:21]: Do you want to be precise about the ort backend here is specified for training?

**Review by fxmarty — APPROVED** [review]
LGTM

File: docs/source/onnxruntime/usage_guides/trainer.mdx (hunk 3 above)
Comment (fxmarty) [reply docs/source/onnxruntime/usage_guides/trainer.mdx:23]: Could we explicit it more then? `...` is not very informative

File: docs/source/onnxruntime/usage_guides/trainer.mdx (hunk 2 above)
Comment (fxmarty) [reply docs/source/onnxruntime/usage_guides/trainer.mdx:21]: `With ONNX Runtime backend` just sound not super good. Maybe with "the"? I don't know

File: docs/source/onnxruntime/usage_guides/trainer.mdx [hunk 8]
```diff
@@ -10,14 +10,43 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+and [AMD](https://cloudblogs.microsoft.com/opensource/2021/07/13/onnx-runtime-release-1-8-1-previews-support-for-accelerated-training-on-amd-gpus-with-the-amd-rocm-open-software-platform/) GPUs.
+With ONNX Runtime backend, [`ORTTrainer`] and [`ORTSeq2SeqTrainer`] take advantage of:
+
+* Computation graph optimizations: node fusions, node eliminations...
+* Efficient memory planning
+* Kernel optimization
+* ORT fused Adam optimizer: batches the elementwise updates applied to all the model's parameters into one or a few kernel launches
+* More efficient FP16 optimizer: eliminates a great deal of device to host memory copies
+* Mixed precision training
+* ...
```
Comment (fxmarty) [inline docs/source/onnxruntime/usage_guides/trainer.mdx:29]:
```suggestion
```

File: docs/source/onnxruntime/usage_guides/trainer.mdx (hunk 7 above)
Comment (JingyaHuang) [reply docs/source/onnxruntime/usage_guides/trainer.mdx:38]: No doc from ORT' side, it is available in `ORTTrainingArguments`.

Comment (fxmarty) [reply docs/source/onnxruntime/usage_guides/trainer.mdx:38]: 🫠

File: docs/source/onnxruntime/usage_guides/trainer.mdx (hunk 2 above)
Comment (JingyaHuang) [reply docs/source/onnxruntime/usage_guides/trainer.mdx:21]: Added.

File: docs/source/onnxruntime/usage_guides/trainer.mdx (hunk 1 above)
Comment (JingyaHuang) [reply docs/source/onnxruntime/usage_guides/trainer.mdx:16]: Looks good.

**Review by regisss — APPROVED** [review]
Great addition to the doc :rocket: 
You're waiting for the blog post to be released to merge this PR?

**Comment by JingyaHuang** [comment]
@regisss Yes, I'm going to merge the PR this afternoon just before the release of the blog post.

**Pull request merged**